numpy/core/tests/test_multiarray.py

```diff
@@ -5495,6 +5495,33 @@ def test_mean_axis_error(self):
         with assert_raises(np.core._exceptions.AxisError):
             np.arange(10).mean(axis=2)
 
+    def test_mean_where(self):
+        a = np.arange(16).reshape((4, 4))
+        wh_full = np.array([[False, True, False, True],
+                            [True, False, True, False],
+                            [True, True, False, False],
+                            [False, False, True, True]])
+        wh_partial = np.array([[False],
+                               [True],
+                               [True],
+                               [False]])
+        tgt = np.array([2.0, 5.0, 8.5, 14.5])
+        assert_allclose(a.mean(axis=1, where=True),
+                            np.array([1.5, 5.5, 9.5, 13.5]))
+        assert_allclose(a.mean(axis=0, where=wh_full),
+                            np.array([6., 5., 10., 9.]))
+        assert_allclose(a.mean(axis=1, where=wh_full),
+                            np.array([2., 5., 8.5, 14.5]))
+        assert_allclose(a.mean(axis=0, where=wh_partial),
+                            np.array([6., 7., 8., 9.]))
+        
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter('always')
```

Comment:
I think you can use `with pytest.warns(RuntimeWarning) as w` here

Reply:
You're right, now it's more concise. Is there a special reason why in other tests, `warnings.catch_warnings` is used?

Reply:
We didn't use to have a pytest dependency.